src/meshes/p4est_mesh.jl

```diff
@@ -11,7 +11,8 @@
 An unstructured curved mesh based on trees that uses the C library `p4est`
 to manage trees and mesh refinement.
```

Comment:
You suffer a bit from facing a relatively sparse documentation of the `P4estMesh` in the first place, but maybe this is now a good time to add a couple more lines what is going on here.
At least `NDIMS_AMBIENT` needs to be mentioned in the docstring.

Reply:
Thanks @DanielDoehring for pointing this out! I have described the meaning of `NDIMS` and `NDIMS_AMBIENT` in the docstring now. I agree that the documentation of the `P4estMesh` type itself is sparse, although the constructors which users typically would use are fairly well documented. We will also add documentation in TrixiAtmo.jl about discretizations on manifolds etc., so that can be linked here eventually.

Reply:
True, what I meant with documentation would be something more developer-faced, i.e., just some couple notes what is going on here. Sorry for not being precise in the first place! I like the current version :+1: